x/evm/simulation/operations.go

```diff
@@ -0,0 +1,306 @@
+		return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgEthereumTx, "can not sign ethereum tx"), nil, err
+	}
+
+	_, _, err = ctx.bapp.Deliver(txConfig.TxEncoder(), signedTx)
+	if err != nil {
+		return simtypes.NoOpMsg(types.ModuleName, types.TypeMsgEthereumTx, "failed to deliver tx"), nil, err
+	}
+
+	return simtypes.OperationMsg{}, fops, nil
+}
+
+// Populate the ethereum tx with valid random values
+func CreateRandomValidEthTx(ctx *simulateContext, from, to *common.Address, amount *big.Int, data *hexutil.Bytes) (ethTx *types.MsgEthereumTx, err error) {
+	estimateGas, err := EstimateGas(ctx, from, to, data)
+	if err != nil {
+		return nil, err
+	}
+	gasLimit := estimateGas + uint64(ctx.rand.Intn(int(sdktx.MaxGasWanted-estimateGas)))
+	ethChainID := ctx.keeper.ChainID()
+	chainConfig := ctx.keeper.GetParams(ctx.context).ChainConfig.EthereumConfig(ethChainID)
+	gasPrice := ctx.keeper.BaseFee(ctx.context, chainConfig)
+	gasFeeCap := new(big.Int).Add(gasPrice, big.NewInt(int64(ctx.rand.Int())))
+	gasTipCap := big.NewInt(int64(ctx.rand.Int()))
+	nonce := ctx.keeper.GetNonce(ctx.context, *from)
+
+	if amount == nil {
+		amount, err = RandomTransferableAmount(ctx, *from, gasLimit, gasFeeCap)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	ethTx = types.NewTx(ethChainID, nonce, to, amount, gasLimit, gasPrice, gasFeeCap, gasTipCap, *data, nil)
+	ethTx.From = from.String()
+	return ethTx, nil
+}
+
+// Estimate gas used by quering the keeper
+func EstimateGas(ctx *simulateContext, from, to *common.Address, data *hexutil.Bytes) (gas uint64, err error) {
+	args, err := json.Marshal(&types.TransactionArgs{To: to, From: from, Data: data})
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := ctx.keeper.EstimateGas(sdk.WrapSDKContext(ctx.context), &types.EthCallRequest{
+		Args:   args,
+		GasCap: config.DefaultGasCap,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return res.Gas, nil
+}
+
+// Transferable amount is between the range [0, spendable), spendable = balance - gasFeeCap * GasLimit
+func RandomTransferableAmount(ctx *simulateContext, address common.Address, gasLimit uint64, gasFeeCap *big.Int) (amount *big.Int, err error) {
+	balance := ctx.keeper.GetBalance(ctx.context, address)
+	feeLimit := new(big.Int).Mul(gasFeeCap, big.NewInt(int64(gasLimit)))
+	if (feeLimit.Cmp(balance)) > 0 {
+		return nil, ErrNoEnoughBalance
+	}
+	spendable := new(big.Int).Sub(balance, feeLimit)
+	if spendable.Cmp(big.NewInt(0)) == 0 {
+		amount = new(big.Int).Set(spendable)
+		return amount, nil
+	}
+	simAmount, err := simtypes.RandPositiveInt(ctx.rand, sdk.NewIntFromBigInt(spendable))
+	if err != nil {
+		return nil, err
+	}
+	amount = simAmount.BigInt()
+	return amount, nil
+}
+
+// TxConfig helps with pack the ethereum tx and deliver it.
+func NewTxConfig() client.TxConfig {
+	interfaceRegistry := codectypes.NewInterfaceRegistry()
+	marshaler := codec.NewProtoCodec(interfaceRegistry)
+	txConfig := tx.NewTxConfig(marshaler, tx.DefaultSignModes)
+	return txConfig
+}
```

Comment:
I think we can remove this and use the one from the SDK config (`sdk.GetConfig`)